### DISCUSSION TITLE
请问E5的是不是搭建不成功？

### DISCUSSION THREAD

**Discussion by xxllllll** [spencerwooo/onedrive-vercel-index#70] [Q&A] [answered]
大佬你好，用E5的搭建完显示Error: Request failed with status code 500.不知道为什么。
Vercel中显示的error为
[GET] /api?path=/
12:50:12:43
AAAAD--DLA3VO7QrddgJg7WevrAgDs_wQA9P9vMoeGIJz0OZdWzXpb34tvqW50cREH8ef_duBLVxq_Q9swV3hDte8GuQxGGAPa64karLjCrYVFFtKPPLHzbLRDkTA_QZfRRE9EmqRWSRGzRfLXkKgKJxUlsuULmv-djw3ytMDlMnegtUTJGga0mpMDZYUT1b6hxSuSmNkJIxFztOl_DbLVn4CQjC4IGLiy9nsQEpl5GRwgIpXYSJOcIRSDRwz4Yhbh3mEZ72AS57ih9UNLGwO9XtUlxH2a-jMRSdTAMGqtXCt8oiOlMDZuICX9hBHf_e0fZYv9qKImZt_TEL9wc5Vc1ytT2_b0jvrnTqKL0zH02Ex8CrlLz6ODr1T4glyb0LvPuw2BEFjTBYLN3m8s7x6Oy9QorENv8hqoGKOli2pO_V7FweP29DkW_Q_2X6a7MirSSySBxLdhfvY3ULG_TTOF5532ojZqOnViaujTz23vaN-WcnzigKpY85Un0BMkD6O7aA28HmxXpo5vebLH9rC2TS4-2KHV96V6VuhRSMXfobBqyjmJJa-jdwA0E10HdXa51WIC93MtoPPxA0rGZV1QAWU88e1F2XgPyfn3M7iUDNdMQPCW_hJnzdtF-F7yUUMZuNSmZqF5o7v0EsbAcmGzn7zHRaX57SYi5ZkA2gSb_P2_j0o7nBy5oQ8jeNgax6qb-Trmkq3TCh52R1pGEu-KNWGlXNkUeIeQ0uUwmq6ZH9nUUpOCaTJhRJ84SlBwmshaS7Ln2FKfkFER9dvrbiKw0em-tGDznNnjTzMux42W9yN-Gqi1otcLKaHUk1g8nwxl1H6knbf9d18m-VBXCVPeOoT7BnZna0rXzqWQ5uwQg7b4A2fwFfOt6A1-TabDqj7Go3pUtWihlLeCoFWMQshHecjW4tJ0qIgkNtn4SePe8i1Ng32uS-afLSyI7j0hkUfA4G3Dw_0PAk_kPbBs9MNv0HM&grant_type=refresh_token',
      headers: [Object],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      adapter: [Function: httpAdapter],
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      validateStatus: [Function: validateStatus]
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'POST /common/oauth2/v2.0/token HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/x-www-form-urlencoded\r\n' +
        'User-Agent: axios/0.21.1\r\n' +
        'Content-Length: 1216\r\n' +
        'Host: login.microsoftonline.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/common/oauth2/v2.0/token',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'login.microsoftonline.com',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: {
      error: 'invalid_client',
      error_description: "AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application: '54789609-d25f-450b-b312-3de409813c18'.\r\n" +
        'Trace ID: 2ecb3204-dc82-4048-ac8d-1b21beb7a300\r\n' +
        'Correlation ID: c10dcc94-2de4-47b2-ad53-1b6a1635ca46\r\n' +
        'Timestamp: 2021-08-31 04:50:13Z',
      error_codes: [Array],
      timestamp: '2021-08-31 04:50:13Z',
      trace_id: '2ecb3204-dc82-4048-ac8d-1b21beb7a300',
      correlation_id: 'c10dcc94-2de4-47b2-ad53-1b6a1635ca46',
      error_uri: 'https://login.microsoftonline.com/error?code=50011'
    }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
RequestId: 9a531056-b30f-4adb-9e2d-f002de21ade4 Error: Runtime exited with error: exit status 1
Runtime.ExitError

**Comment by spencerwooo** [comment, marked as answer]
```
AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application: '54789609-d25f-450b-b312-3de409813c18'.
```

看起来是 redirect url 的配置问题，可能可以成功部署。

**Comment by sldarkwd** [comment]
redirect url 即 replay URL
可能需要在 Active Directory-应用注册-“Client_id对应的应用程序”-身份验证-平台配置-WEB 处添加URI
与config/api.config.js中的redirectUri保持一致
参考
https://docs.microsoft.com/zh-cn/azure/active-directory/develop/reply-url

**Comment by spencerwooo** [comment]
这个好久之前的问题了，现版本我确认 E5 开发者版本是可以搭建的。

**Reply by spencerwooo** [reply]
另外要是用项目提供的 clientId 和 clientSecret 也不必考虑 redirectUri 的事情了。

**Reply by zkeq** [reply]
我的 E5 也是可以的

今天上午才换上
